### PR TITLE
Use frames count instead of delay to check animated GIFs

### PR DESCRIPTION
**Pull request by nbianca**
There are GIFs with multiple frames and zero delay, but also GIFs with
a single frame and positive delay. The old algorithm failed for this
kind of images.

### CODE REVIEW

**Comment by Nakilon** [comment]
That's what I said: https://github.com/sdsykes/fastimage/pull/114#issuecomment-630255884
No need to do assumptions about whether it's animated or not -- just give library user the number of frames.

File: lib/fastimage.rb [hunk 1]
```diff
@@ -578,38 +578,45 @@ def width_and_height
...
+          @stream.skip(1) # LZW min code size (1)
           skip_sub_blocks
         else
-          return # unrecognized block
+          break # unrecognized block
         end
       end
 
-      delay > 0
+      frames > 1
```
Comment (sdsykes) [inline lib/fastimage.rb:619]: Isn't it the case that frames can never be greater than 1 here?

Comment (nbianca) [reply lib/fastimage.rb:619]: Yes, if this line is executed it means that `frames > 1` never happened. I can change this to return always false.

**Comment by nbianca** [comment]
@Nakilon, I could do that as well, but it would require parsing the whole GIF image. My understanding is that this library is meant to  provide meta information by "fetching as little as needed".

@sdsykes, can you please review this again? I amended my commit and forced pushed.

**Comment by sdsykes** [comment]
This looks ok!

**Comment by nbianca** [comment]
Hello @sdsykes! Did you set a date when you will cut a new version of this gem?

**Comment by SamSaffron** [comment]
@sdsykes so sorry for being a nag here, any chance you can cut a new version for the festive season?

**Comment by sdsykes** [comment]
Well, a late festive version. I just pushed 2.2.1.

**Pull request merged**